src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,94 @@
+package duke.command;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.Ui;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+
+
+/**
+ * The AddCommand class represents a command to add a task.
+ * It is a subclass of the Command class.
+ */
+public class AddCommand extends Command {
+    private String taskType;
+
+    /**
+     * Constructs an AddCommand object with the provided command details and task type.
+     *
+     * @param commandDetails The list of command details, which may include task information.
+     * @param taskType       The type of task to add (e.g., "T" for Todo, "D" for Deadline, "E" for Event).
+     */
+    public AddCommand(ArrayList<String> commandDetails, String taskType) {
+        super(commandDetails);
+        this.taskType = taskType;
+    }
+
+    @Override
```

Comment:
I think can still write a javadoc for these execute statements since they have differing behaviours across the different classes! :)